includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -992,34 +996,38 @@ public function save_source_to_order( $order, $source ) {
 	 * @param int    $balance_transaction_id
 	 */
 	public function update_fees( $order, $balance_transaction_id ) {
-		$balance_transaction = WC_Stripe_API::retrieve( 'balance/history/' . $balance_transaction_id );
+		try {
+			$balance_transaction = WC_Stripe_API::retrieve( 'balance/history/' . $balance_transaction_id );
 
-		if ( empty( $balance_transaction->error ) ) {
-			if ( isset( $balance_transaction ) && isset( $balance_transaction->fee ) ) {
-				// Fees and Net needs to both come from Stripe to be accurate as the returned
-				// values are in the local currency of the Stripe account, not from WC.
-				$fee_refund = ! empty( $balance_transaction->fee ) ? WC_Stripe_Helper::format_balance_fee( $balance_transaction, 'fee' ) : 0;
-				$net_refund = ! empty( $balance_transaction->net ) ? WC_Stripe_Helper::format_balance_fee( $balance_transaction, 'net' ) : 0;
+			if ( empty( $balance_transaction->error ) ) {
+				if ( isset( $balance_transaction ) && isset( $balance_transaction->fee ) ) {
+					// Fees and Net needs to both come from Stripe to be accurate as the returned
+					// values are in the local currency of the Stripe account, not from WC.
+					$fee_refund = ! empty( $balance_transaction->fee ) ? WC_Stripe_Helper::format_balance_fee( $balance_transaction, 'fee' ) : 0;
+					$net_refund = ! empty( $balance_transaction->net ) ? WC_Stripe_Helper::format_balance_fee( $balance_transaction, 'net' ) : 0;
 
-				// Current data fee & net.
-				$fee_current = WC_Stripe_Helper::get_stripe_fee( $order );
-				$net_current = WC_Stripe_Helper::get_stripe_net( $order );
+					// Current data fee & net.
+					$fee_current = WC_Stripe_Helper::get_stripe_fee( $order );
+					$net_current = WC_Stripe_Helper::get_stripe_net( $order );
 
-				// Calculation.
-				$fee = (float) $fee_current + (float) $fee_refund;
-				$net = (float) $net_current + (float) $net_refund;
+					// Calculation.
+					$fee = (float) $fee_current + (float) $fee_refund;
+					$net = (float) $net_current + (float) $net_refund;
 
-				WC_Stripe_Helper::update_stripe_fee( $order, $fee );
-				WC_Stripe_Helper::update_stripe_net( $order, $net );
+					WC_Stripe_Helper::update_stripe_fee( $order, $fee );
+					WC_Stripe_Helper::update_stripe_net( $order, $net );
 
-				$currency = ! empty( $balance_transaction->currency ) ? strtoupper( $balance_transaction->currency ) : null;
-				WC_Stripe_Helper::update_stripe_currency( $order, $currency );
+					$currency = ! empty( $balance_transaction->currency ) ? strtoupper( $balance_transaction->currency ) : null;
+					WC_Stripe_Helper::update_stripe_currency( $order, $currency );
 
-				if ( is_callable( [ $order, 'save' ] ) ) {
-					$order->save();
+					if ( is_callable( [ $order, 'save' ] ) ) {
+						$order->save();
+					}
 				}
+			} else {
+				WC_Stripe_Logger::log( 'Unable to update fees/net meta for order: ' . $order->get_id() );
 			}
-		} else {
+		} catch ( Exception $e ) {
```

Comment:
Here, the following high-level methods expect this exception:
- WC_REST_Stripe_Orders_Controller::capture_terminal_payment * → capture_payment
- WC_Gateway_Stripe::process_payment * → process_response